azureLogIngestion/adapter/opentelemetry/index.ts

```diff
@@ -24,9 +36,39 @@ export class OpenTelemetryAdapter {
         this.traceProvider.register()
     }
 
-    addSpan(): void {
-        const span = this.traceProvider.getTracer("default").startSpan("foo")
-        span.setAttribute("key", "value")
+    private convertContext(appSpan: Record<string, any>, ctx: Context): any {
+        let obj: { traceId: string; spanId: string; traceFlags: TraceFlags; traceState: any }
+        // eslint-disable-next-line prefer-const
+        obj = {
+            traceId: appSpan.parentId,
+            spanId: appSpan.id,
+            traceFlags: 0,
+            traceState: ctx.traceContext.tracestate,
+        }
+        return obj
+    }
+
+    addSpan(appSpan: Record<any, any>, context: Context): void {
+        const span = opentelemetry.trace.getTracer("default").startSpan(appSpan.name)
```

Comment:
This  should be `this.traceProvider.getTracer('default')` to ensure we're using the class instance.